controllers/dataplane_controller.go

```diff
@@ -85,6 +91,14 @@ func (r *DataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 		return ctrl.Result{}, nil // no need to requeue, the update will trigger.
 	}
+	// validate dataplane
+	v := dataplanevalidation.NewValidator(r.Client)
+	err = v.Validate(dataplane)
+	if err != nil {
+		debug(log, "failed to validate dataplane: error "+err.Error(), dataplane)
```

Comment:
across KIC and other codebases of ours, it's pretty common to just put the error itself after the colon, if that sounds reasonable to you:

```suggestion
		debug(log, "failed to validate dataplane: "+err.Error(), dataplane)
```